File: wrapUpSnacks/python/wrap_up_snacks.py
def factorial_of(number):
    factorial = 1
    for each_number in range(number, 0, -1):
        factorial *= each_number
    return factorial
    
def greatest_common_divisor_of(first_number, second_number):
    divisor = 1
    common_divisor = 0
    while divisor <= first_number or divisor <= second_number :
        if first_number % divisor == 0 and second_number % divisor == 0 :
            common_divisor = divisor;
        divisor += 1
    return common_divisor
    
def lowest_common_multiple_of(first_number, second_number) :
    common_divisor = greatest_common_divisor_of(first_number, second_number)
    return (first_number * second_number) / common_divisor

def is_perfect_number(number):
    total = 0;
    factor = 1
    while factor < number :
        if number % factor == 0 :
            total += factor
        factor += 1
    return total == number
    
def count_number_of_digits_in(number) :
    digits_number = 0
    digit = 0
    while number != 0 :
        digit = number % 10
        digits_number += 1
        number //= 10
    return digits_number
    
def is_armstrong(number):
    given_number = number
    total = 0
    exponient = count_number_of_digits_in(number)
    while number != 0 :
        digit = number % 10
        total += digit ** exponient
        number //= 10
    return total == given_number

def is_strong(number):
    given_number = number
    total = 0
    while number != 0 :
        digit = number % 10
        total += factorial_of(digit)
        number //= 10
    return total == given_number

def is_leap_year(year) :
    if year % 4 == 0 :
        if year % 100 != 0 or year % 400 == 0 :
            return True 
    return False

File: wrapUpSnacks/javaScript/WrapUpSnacks.js

function factorialOf(number) {
    let factorial = 1;
    for(let eachNumber = number; eachNumber >= 1; eachNumber--){
        factorial *= eachNumber
    }
    return factorial
}
const greatestCommonDivisorOf = function(firstNumber, secondNumber) {
    let divisor = 1
    let commonDivisor = 0;
    while (divisor <= firstNumber || divisor <= secondNumber) {
        if(firstNumber % divisor === 0 && secondNumber % divisor === 0) {
            commonDivisor = divisor
        }
        divisor++
    }
    return commonDivisor
}
const lowestCommonMultipleOf = function(firstNumber, secondNumber) {
    let commonDivisor = greatestCommonDivisorOf(firstNumber, secondNumber)
    return (firstNumber * secondNumber) / commonDivisor
}
function isPerfectNumber(number) {
    let total = 0
    let factor = 1
    while (factor < number) {
        if(number % factor == 0){
            total += factor
        }
        factor++
    }
    return total === number
}
const numberOfDigitsIn = function(number) {
    let count = 0
    while (number != 0) {
        let digit = number % 10
        count++
        number = Math.floor(number / 10)
    }
    return count
}
const isArmstrong = function(number) {
    const givenNumber = number
    let total = 0
    let exponient = numberOfDigitsIn(number)
    while(number != 0){
        let digit = number % 10
        total += digit ** exponient
        number = Math.floor(number / 10)
    }
    return total === givenNumber
}
const isStrong = function(number) {
    const givenNumber = number
    let total = 0
    while (number != 0){
        let digit = number % 10
        total += factorialOf(digit)
        number = Math.floor(number / 10)
    }
    return total === givenNumber
}
const isLeapYear = function(year) {
    if(year % 4 === 0) {
        if(year % 100 !== 0 || year % 400 === 0) {
            return true
        }
    }
    return false
}
console.log(factorialOf(5))
console.log(greatestCommonDivisorOf(18,12))
console.log(lowestCommonMultipleOf(18,12))
console.log(isPerfectNumber(6))
console.log(numberOfDigitsIn(416))
console.log(isArmstrong(153))
console.log(isStrong(145))
console.log(isLeapYear(2000))
console.log()
for(let number = 1; number <= 1000; number++){
    if(isArmstrong(number)) {
        console.log("Armstrong number --> " + number)
    }
}
console.log()
for(let number = 1; number <= 1000; number++){
    if(isPerfectNumber(number)) {
        console.log("Perfect number --> " + number)
    }
}
console.log()
for(let number = 1900; number <= 2025; number++){
    if(isLeapYear(number)) {
        console.log("Leap year --> " + number)
    }
}
console.log()
let leapYearCount = 0
for(let number = 1900; number <= 2025; number++){
    if(isLeapYear(number)) {
        leapYearCount++;
    }
}
console.log("The number of leap year between 1900 to 2025 is: " + leapYearCount)
